### ISSUE TITLE
org-jira-get-issues end with "error in process sentinel"

### ISSUE THREAD

**Issue by dod38fr** [ahungry/org-jira#352]
Hi

I'm trying to use `org-jira` with a private JIRA instance managed by my client. 

`org-jira-get-issues` command fails with:
```
About to render 17 issues.
(New file)
error in process sentinel: if: End of buffer
error in process sentinel: End of buffer
Auto-saving...done
```

Any idea on how to fix this ?

**Comment by damien-biasotto** [comment]
I have the same issue running on `GNU Emacs 31.0.50 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.42, cairo version 1.18.0)` with org `Org mode version 9.7.6 (9.7.6-7a4527)` and running org-jira `20230915.57`

**Comment by ahungry** [comment]
@damien-biasotto are you also using a privately hosted jira instance?

org-jira as a package hasn't had a change in almost a year, so it seems something else (outside of the package) may have, or the change is the instances being used.

**Comment by gasquid** [comment]
Not sure if this is helpful, but I had pinned org to an older version (something I found on Reddit) to work around another issue. When I upgraded org, I had the same issue. I'm using Jira Cloud.

From my packages.el:
(package! org :pin "28dc5dd1ac1325ecb6bdec3e9f72997c35eebfa0")

**Comment by damien-biasotto** [comment]
> @damien-biasotto are you also using a privately hosted jira instance?
> 
> org-jira as a package hasn't had a change in almost a year, so it seems something else (outside of the package) may have, or the change is the instances being used.

I'm also using a JIRA Cloud instance (and using API Token for authentication). I will try to use a downgraded version of org and see if it helps.

Thanks for the prompt response @ahungry  and thanks for the workaround @gasquid :)

**Comment by damien-biasotto** [comment]
Ok downgrading my org-mode to `9.6.30` fixed the issue for me aswell. I will try to pinpoint when the regression has been introduced

**Comment by dod38fr** [comment]
To close this issue, I'd rather wait for the issue to be fixed without pinning org-mode version.

**Comment by ahungry** [comment]
https://orgmode.org/Changes.html

Looks like `org-insert-subheading` behavior was changed - maybe that's the culprit?

We use it in 3 locations in org-jira.el - in org 9.7.4 it is defined as such:

```
;; Defined in /usr/local/share/emacs/30.0.50/lisp/org/org.el.gz
(defun org-insert-subheading (arg)
  "Insert a new subheading and demote it.
Works for outline headings and for plain lists alike.
The prefix argument ARG is passed to `org-insert-heading'.
Unlike `org-insert-heading', when point is at the beginning of a
heading, still insert the new sub-heading below."
  (interactive "P")
  (when (bolp) (forward-char))
  (org-insert-heading arg)
  (cond
   ((org-at-heading-p) (org-do-demote))
   ((org-at-item-p) (org-indent-item))))
```

Could someone on 9.6 please share the definition?  I'm guessing we will need to put the pre 9.7 version of the function directly in the org-jira.el file and change references to use it, since apparently it's not a stable interface/behavior, despite being defined as a public function in org.

**Comment by ahungry** [comment]
Linked issue addressed this a few weeks back (though I missed it until now) - I think since it only seems to happen when preceded by `(goto-char (point-max))` (as noted in linked issue), the simplest fix may be opening the newline as recommended in #351 - although I don't know if that would break up rendering order on 9.6 and prior orgs (if someone on 9.6 could try testing out that fix - the diff should be:

```
diff --git a/org-jira.el b/org-jira.el
index 0f45c41..9b0f4ff 100644
--- a/org-jira.el
+++ b/org-jira.el
@@ -1172,6 +1172,7 @@ ORG-JIRA-PROJ-KEY-OVERRIDE being set before and after running."
                                                      (if (org-goto-first-child)
                                                          (org-insert-heading)
                                                        (goto-char (point-max))
+                                                       (open-line 1)
                                                        (org-insert-subheading t))
                                                      (org-jira-insert entry-heading "\n"))
```

**Comment by gasquid** [comment]
The above change appears to be rendering issues for me correctly with org 9.6.21, 9.6.30, and 9.7.5.

**Comment by ahungry** [comment]
Awesome, thanks!  Going to push this and hopefully it should resolve for all.

**Comment by ahungry** [comment]
Please re-open or comment if it still is problematic - fix pushed in ac625b0

**Comment by dod38fr** [comment]
Also working fine here ! 

Thanks for the quick fix :smiley:
